src/libcollections/string.rs

```diff
@@ -701,6 +701,13 @@ impl String {
     /// Violating these may cause problems like corrupting the allocator's
     /// internal datastructures.
     ///
+    /// The ownership of `ptr` is effectively transferred to the
+    /// `String` and the memory referred to by the pointer will be
+    /// deallocated when the `String` goes out of scope. Ensure that
+    /// nothing else uses the pointer after calling this
+    /// function. Calling this function multiple times with the same
+    /// pointer is an error.
```

Comment:
> if we begin realloc-ing the pointer in question

or even this sequence:

```rust
let mut s = String::from_raw_parts(x, ...);
s.push("I'm reallocating");
mem::forget(s);
let mut s = String::from_raw_parts(x, ...);
```

> String then may deallocate, reallocate or change the contents of memory pointed to by the pointer at will.

I like this. I think it does a good job of covering the important aspects that you can't really trust the pointer anymore.